src/standalone/diagnostics/reservedFileNameDiagnostics.node.ts

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { inject, injectable } from 'inversify';
+import {
+    CancellationToken,
+    CodeAction,
+    CodeActionContext,
+    CodeActionKind,
+    CodeActionProvider,
+    commands,
+    ConfigurationTarget,
+    Diagnostic,
+    DiagnosticCollection,
+    DiagnosticSeverity,
+    EventEmitter,
+    FileDecoration,
+    FileDecorationProvider,
+    languages,
+    Range,
+    TextDocument,
+    TextEditor,
+    ThemeColor,
+    Uri,
+    window,
+    workspace
+} from 'vscode';
+import { IDisposable } from '@fluentui/react';
+import { IExtensionSingleActivationService } from '../../platform/activation/types';
+import { PYTHON_LANGUAGE } from '../../platform/common/constants';
+import { disposeAllDisposables } from '../../platform/common/helpers';
+import { Common, DataScience } from '../../platform/common/utils/localize';
+import * as path from '../../platform/vscode-path/path';
+import { IFileSystem } from '../../platform/common/platform/types';
+import { IWorkspaceService } from '../../platform/common/application/types';
+import { IReservedPythonNamedProvider } from '../../platform/interpreter/types';
+import { JupyterKernelStartFailureOverrideReservedName } from '../../platform/interpreter/constants';
+import { swallowExceptions } from '../../platform/common/utils/decorators';
+
+export const enabledSettingName = 'diagnostics.reservedPythonNames.enabled';
```

Comment:
This is a bit funny to be just floating loose as an export. For testing?

Reply:
Yes thats correct, do you think we should not export these and just hard code this value in the tests as well?
I used to not export and just hard code in the tests, this way its not exported unnecessarily and the tests also verify whether we're using the right values, however over time i've changed to just export it.

Thoughts?

Reply:
Not sure that I see it as a big deal. Since the tests have the class I guess having it as a public static could be a possibility.

ReservedFileNamesDiagnosticProvider.enabledSettingName

Constraints it a bit to the class in question. But it's already scoped via the file it's in, so again I'm fine with it as is.

Reply:
Agreed, but not a fan of statics as well, as that's not really a const, even though we could create a `static readonly`
leaving this as is.